gammapy/image/core.py

```diff
@@ -1088,3 +1091,82 @@ def assert_allclose(image1, image2, check_wcs=True):
             assert_wcs_allclose(image1.wcs, image2.wcs)
         else:
             raise ValueError('One image has `wcs==None` and the other does not.')
+
+    def convolve(self, kernel, fft=False, mode='reflect', **kwargs):
+        """
+        Convolve sky image with kernel.
+
+        Parameters
+        ----------
+        kernel : `~astropy.convolution.Kernel`
+            Convolution kernel.
+        fft : bool
+            Use fast FFT convolution.
+        mode : str ('reflect')
+            Boundary mode passed to `~numpy.pad`
+        **kwargs : dict
+            Further keyword arguments passed to `~numpy.pad`.
+        """
+        from scipy.signal import fftconvolve
+        from scipy.ndimage import convolve
+
+        width = kernel.array.shape[0] / 2
+        padded = self.pad(pad_width=(width, width))
+
+        if fft:
+            data = fftconvolve(padded.data, kernel.array, mode='same')
+        else:
+            data = convolve(padded.data, kernel.array, mode='constant')
+
+        convolved = self.__class__(data=data, wcs=self.wcs.copy())
+        return convolved.crop(crop_width=(width, width))
+
+    def __mul__(self, image):
+        """
+        Multiply sky image data.
+        """
+        # TODO: check if wcs are compatible, if not raise error
```

Comment:
I'm 👎  on adding `__mul__` and other operators on `SkyImage`.
It's not really needed, if the user wants to do an operation with `data`, she can type `image.data`.

If we go down this route, it would eventually grow and become a ton of code and complex:
already you're thinking about WCS checking, next someone notices that mutliplying with a number or quantity doesn't work, next someone notices that `SkyCube` doesn't support this.

Quickly we would have 1000 lines or code and 1000 lines of tests to implement such operators for many classes in Gammapy (spectra, images, IRFs, ...).

So 👍  to remove and show users via docs examples how to work with `SkyImage.data` and `SkyImage.wcs`.